avocado/utils/process.py

```diff
@@ -791,6 +803,20 @@ def get_pid(self):
         self._init_subprocess()
         return self._popen.pid
 
+    def get_user(self):
+        """
+        Reports user of this process
+        """
+        self._init_subprocess()
+        return get_user_of_pid(self.get_pid())
+
+    def is_running_as_sudo(self):
+        """
+        Returns whether a process is running as sudo
+        """
+        self._init_subprocess()
+        return self.get_user() == 'root'
```

Comment:
Even though this *usually* works, it's not be the condition for an UNIX process to be considered privileged.  The process effective user id being 0 is the definitive check.  What I mean is, you can change the name of your `root` user, but as long as it has uid == 0, it will be able to perform privileged actions.